src/builtins/cd.c

```diff
@@ -1,38 +1,40 @@
 #include <minishell.h>
 
+static int	cd_error_msg(t_status *status, char *msg, bool e_gen)
+{
+	ft_putstr_fd("cd: error ", STDERR_FILENO);
+	ft_putendl_fd(msg, STDERR_FILENO);
+	if (e_gen)
+		*status = E_GENERAL;
+	else
+		*status = E_BUILTIN;
+	return (1);
+}
+
 int	ft_cd(char **arguments, t_minishell *shell)
 {
 	char	*cwd;
 	char	*dir;
 
 	dir = arguments[1];
-	if (dir == NULL)
+	if (!dir)
 	{
 		dir = dict_get(&shell->env, "HOME");
-		if (dir == NULL)
-		{
-			ft_putstr_fd("cd: no home directory found\n", STDERR_FILENO);
-			return (E_BUILTIN);
-		}
+		if (!dir)
+			return (cd_error_msg(&shell->status,
```

Comment:
this one is actually tricky, it could be a memory managment failure or that the dir doesn't exist. It is to complicated to change now so I would just leave it like this